src/sky.cpp

```diff
@@ -202,58 +203,132 @@ void Sky::render()
+					else if (j == 2)
+						// Switch from -Z (south) to -X (west)
+						vertex.Pos.rotateXZBy(-90);
+					else
+						// Switch from -Z (south) to +Z (north)
+						vertex.Pos.rotateXZBy(-180);
+				}
+				driver->drawIndexedTriangleFan(&vertices[0], 4, indices, 2);
+			}
+		} else {
+			// Draw middle band of our smooth skybox
+			for (u32 j = 0; j < 4; j++) {
+				video::SColor c = m_sky_color_bottom;
+				vertices[0] = video::S3DVertex(-1, -0.03, -1, 0, 0, 1, c, t, t);
+				vertices[1] = video::S3DVertex( 1, -0.03, -1, 0, 0, 1, c, o, t);
+				vertices[2] = video::S3DVertex( 1,  0.02, -1, 0, 0, 1, c, o, o);
+				vertices[3] = video::S3DVertex(-1,  0.02, -1, 0, 0, 1, c, t, o);
+				for (video::S3DVertex &vertex : vertices) {
+					if (j == 0)
+						// Don't switch
+						{}
+					else if (j == 1)
+						// Switch from -Z (south) to +X (east)
+						vertex.Pos.rotateXZBy(90);
+					else if (j == 2)
+						// Switch from -Z (south) to -X (west)
+						vertex.Pos.rotateXZBy(-90);
+					else
+						// Switch from -Z (south) to +Z (north)
+						vertex.Pos.rotateXZBy(-180);
+				}
+				driver->drawIndexedTriangleFan(&vertices[0], 4, indices, 2);
+			}
+			
+			// Draw top band of our smooth skybox
+			for (u32 j = 0; j < 4; j++) {
+				video::SColor c = m_sky_color_bottom;
+				video::SColor c_blend = m_sky_color_top;
+				vertices[0] = video::S3DVertex(-1, 0.02, -1, 0, 0, 1, c, t, t);
+				vertices[1] = video::S3DVertex( 1, 0.02, -1, 0, 0, 1, c, o, t);
+				vertices[2] = video::S3DVertex( 1, 0.5, -1, 0, 0, 1, c_blend, o, o);
+				vertices[3] = video::S3DVertex(-1, 0.5, -1, 0, 0, 1, c_blend, t, o);
+				for (video::S3DVertex &vertex : vertices) {
+					if (j == 0)
+						// Don't switch
+						{}
+					else if (j == 1)
+						// Switch from -Z (south) to +X (east)
+						vertex.Pos.rotateXZBy(90);
+					else if (j == 2)
+						// Switch from -Z (south) to -X (west)
+						vertex.Pos.rotateXZBy(-90);
+					else
+						// Switch from -Z (south) to +Z (north)
+						vertex.Pos.rotateXZBy(-180);
+				}
+				driver->drawIndexedTriangleFan(&vertices[0], 4, indices, 2);
 			}
-			driver->drawIndexedTriangleFan(&vertices[0], 4, indices, 2);
-		}
 
-		// Draw far cloudy fog thing
-		for (u32 j = 0; j < 4; j++) {
-			video::SColor c = cloudyfogcolor;
-			vertices[0] = video::S3DVertex(-1, -1.0, -1, 0, 0, 1, c, t, t);
-			vertices[1] = video::S3DVertex( 1, -1.0, -1, 0, 0, 1, c, o, t);
-			vertices[2] = video::S3DVertex( 1, 0.08, -1, 0, 0, 1, c, o, o);
-			vertices[3] = video::S3DVertex(-1, 0.08, -1, 0, 0, 1, c, t, o);
-			for (video::S3DVertex &vertex : vertices) {
-				if (j == 0)
-					// Don't switch
-					{}
-				else if (j == 1)
-					// Switch from -Z (south) to +X (east)
-					vertex.Pos.rotateXZBy(90);
-				else if (j == 2)
-					// Switch from -Z (south) to -X (west)
-					vertex.Pos.rotateXZBy(-90);
-				else
-					// Switch from -Z (south) to +Z (north)
-					vertex.Pos.rotateXZBy(-180);
+			// Draw bottom sides of the sky
```

Comment:
This code block can be combined with `// Draw middle band of our smooth skybox` above since the colour is identical.